packages/tools/src/tools/segmentation/ThresholdPreviewTool.ts

```diff
@@ -0,0 +1,235 @@
+
+import { getEnabledElementByIds } from '@cornerstonejs/core';
+import { LabelmapSegmentationData } from 'tools/src/types/LabelmapTypes';
+
+class ThresholdPreviewTool extends BaseTool {
+  static toolName;
+
+  editData: {
+    annotation: any;
+    segmentationId: string;
+    segmentation: any;
+    imageVolume: any;
+    segmentIndex: number;
+    segmentsLocked: number[];
+    segmentColor: [number, number, number, number];
+    viewportIdsToRender: string[];
+    handleIndex?: number;
+    movingTextBox: boolean;
+    newAnnotation?: boolean;
+    hasMoved?: boolean;
+  } | null;
+  isDrawing: boolean;
+  isHandleOutsideImage: boolean;
+  highlightIntervalId: any;
+
+  constructor(
+    toolProps: PublicToolProps = {},
+    defaultToolProps: ToolProps = {
+      supportedInteractionTypes: ['Mouse', 'Touch'],
+      configuration: {
+        strategies: {
+          FILL_INSIDE_CUBE: fillInsideCube,
+        },
+        strategySpecificConfiguration: {
+          THRESHOLD_INSIDE_CUBE: {
+            threshold: [-150, -70], // E.g. CT Fat // Only used during threshold strategies.
+          },
+        },
+        defaultStrategy: 'FILL_INSIDE_CUBE',
+        activeStrategy: 'FILL_INSIDE_CUBE',
+      },
+    }
+  ) {
+    super(toolProps, defaultToolProps);
+  }
+
+  public getConfiguration = (): any => {
+    const config = segmentationConfig.getGlobalConfig();
+    const { renderInactiveSegmentations } = config;
+
+    const labelmapRepresentationConfig = config.representations.LABELMAP;
+
+    const {
+      renderOutline,
+      outlineWidthActive,
+      renderFill,
+      fillAlpha,
+      fillAlphaInactive,
+      outlineOpacity,
+      outlineOpacityInactive,
+    } = labelmapRepresentationConfig;
+
+    return {
+      fillAlpha,
+      fillAlphaInactive,
+      outlineWidthActive,
+      renderFill,
+      renderInactiveSegmentations,
+      renderOutline,
+      outlineOpacity,
+      outlineOpacityInactive,
+    };
+  };
+
+  setConfiguration = (newConfiguration: Record<string, any>): void => {
+    this.configuration.strategySpecificConfiguration.THRESHOLD_INSIDE_CUBE = {
+      threshold: newConfiguration,
+    };
+
+    this.onSetToolEnabled(null);
+  };
+
+  onSetToolEnabled(event): void {
+    const toolGroupId = this.toolGroupId;
+
+    const toolGroup = getToolGroup(toolGroupId);
+
+    if (!toolGroup) {
+      return;
+    }
+
+    // toolGroup Viewports
+    const toolGroupViewports = toolGroup.viewportsInfo.map(
+      ({ renderingEngineId, viewportId }) => {
+        const enabledElement = getEnabledElementByIds(
+          viewportId,
+          renderingEngineId
+        );
+
+        if (enabledElement) {
+          return enabledElement.viewport;
+        }
+      }
+    );
+
+    const viewport = toolGroupViewports[0];
+
+    const actors = viewport.getActors();
+
+    const firstVolumeActorUID = actors[0].uid;
+    const imageVolume = cache.getVolume(firstVolumeActorUID);
```

Comment:
can you please look at packages/core/src/utilities/getTargetVolumeAndSpacingInNormalDir.ts to see how we find the primary volume? maybe refactor that code inside a util in core